pkg/virt-handler/vm.go

```diff
@@ -414,6 +415,23 @@ func (d *VirtualMachineController) execute(key string) error {
 	return nil
 }
 
+func (d *VirtualMachineController) injectHostDisks(vmi *v1.VirtualMachineInstance) error {
+	// host disks are created based on PersistenVolumeClaims
+	for i := range vmi.Spec.Volumes {
+		if volumeSource := &vmi.Spec.Volumes[i].VolumeSource; volumeSource.PersistentVolumeClaim != nil {
+			pvcSize, err := hostdisk.GetPVCSize(volumeSource.PersistentVolumeClaim.ClaimName, vmi.Namespace, d.clientset)
+			if err != nil {
+				return err
+			}
+			volumeSource.HostDisk = v1.HostDisk{
```

Comment:
If you model the HostDisk like mentioned above, you can replace the PVC with the capacity of the PV, derive the disk-path based on the VMI definition and set the type to `DiskOrCreate`.

Reply:
Ok I will do it.